tenseal/tensealencoder.h

```diff
@@ -25,6 +26,41 @@ class TenSEALEncoder {
         return std::any_cast<std::shared_ptr<T>>(_encoders[tidx]);
     }
 
+    /*
+    Template encoding functions to choose between the use of
+    Integer/BatchEncoder or CKKSEncoder.
+    */
+    template <class T>
+    void encode(std::vector<int64_t>& vec, Plaintext& pt) {
+        auto encoder = this->get<T>();
+        encoder->encode(vec, pt);
+    }
+
+    template <class CKKSEncoder>
+    void encode(std::vector<double>& vec, Plaintext& pt, double scale = 0) {
+        if (scale == 0) scale = pow(2, 40);  // TODO: get from TenSEALContext
```

Comment:
TenSEALEncoder can be created using TenSEALContext instead of SEALContext so that we can fetch global_scale here